### DISCUSSION TITLE
Generate presigned S3 url?

### DISCUSSION THREAD

**Discussion by mrichman** [awslabs/aws-sdk-rust#750] [Q&A] [answered]
I see module `aws_sdk_s3::presigning` but I can't find any examples of how to actually generate a presigned URL from an S3 object.

**Comment by jdisanti** [comment, marked as answer]
We have a couple of examples:

- https://github.com/awslabs/aws-sdk-rust/blob/main/examples/s3/src/bin/get-object-presigned.rs
- https://github.com/awslabs/aws-sdk-rust/blob/main/examples/s3/src/bin/put-object-presigned.rs

So that we can improve discoverability, what was the first place you thought of to look for these examples?

**Reply by urkle** [reply]
@jdisanti I eventually found those as well. And was lost as @mrichman was initially.  I went straight to the aws_sdk_s3::presigning module and found nothing in there for doing presigning :)   then went to the Client and saw no presigning. 

So probably putting some references in the docs for the presigning module to point to the Client#get_object and Client#put_object methods and/or the fluent builders would be helpful.

Now that I have found the GET and PUT presigning. I have one last question.. Where is the presigned POST helper? That right now is what I am looking for.

**Reply by jdisanti** [reply]
Thank you for illustrating the path you took to find the docs! That's very helpful, and I posted a PR to make it a little more discoverable (https://github.com/awslabs/smithy-rs/pull/2903).

There's currently no out of box support for POST Object. I tried creating a workaround for it, but wasn't able to get it working. I've filed the following to track implementing this support: https://github.com/awslabs/aws-sdk-rust/issues/863

**Comment by arcstur** [comment]
They have moved it:

https://github.com/awslabs/aws-sdk-rust/blob/main/examples/examples/s3/src/bin/get-object-presigned.rs
https://github.com/awslabs/aws-sdk-rust/blob/main/examples/examples/s3/src/bin/put-object-presigned.rs

But I wonder why they don't appear [here](https://docs.aws.amazon.com/sdk-for-rust/latest/dg/rust_s3_code_examples.html).

**Reply by DavidSouther** [reply]
It looks like they didn't get registered in https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/.doc_gen/metadata/s3_metadata.yaml which drives the guide landing pages. I've opened https://github.com/awsdocs/aws-doc-sdk-examples/issues/5356 in the docs side.
